### PR TITLE
Make cluster e2e non docker specific.

### PR DESCRIPTION
**Pull request by Random-Liu**
Fixes https://github.com/kubernetes/kubernetes/issues/57979.

This PR:
1) Install `crictl` for GCE GCI/Ubuntu test cluster.
2) Change cluster e2e test to use `crictl`. If `crictl` is not installed, it will fall back to `docker` CLI.

Please note that, this PR:
1) Only installs `crictl` for GCI/Ubuntu image on GCE. If other image maintainers want to support this, it's very easy to install `crictl` `v1.0.0-alpha.0`:
```console
$ wget -O crictl https://storage.googleapis.com/kubernetes-release/crictl/crictl-v1.0.0-alpha.0-linux-amd64
$ chmod +x crictl
// The sha1sum is `075190a36a03beb08065f279693f3e369e70cf99`.
```
2) Only installs `crictl` for test cluster. We may want to install it by default in production cluster after `crictl` is beta in v1.10.

@yujuhong @feiskyer @mrunalp @mikebrow 
/cc @dchen1107 for approval
@kubernetes/sig-node-pr-reviews @kubernetes/sig-testing-bugs 

Signed-off-by: Lantao Liu <lantaol@google.com>

**Release note**:

```release-note
none
```

### CODE REVIEW

**Review by feiskyer — APPROVED** [review]
/lgtm

**Review by mikebrow — COMMENTED** [review]
The code changes for ensuring crictl is available for GCE all look good. I don't see: "Change cluster e2e test to use crictl. If crictl is not installed, it will fall back to docker CLI."

**Comment by Random-Liu** [comment]
@mikebrow It's in test/e2e/framework/utils.go

**Comment by mikebrow** [comment]
@Random-Liu I only see calls for the ps command (health check) being updated.. I don't see "Change cluster e2e test to use crictl. If crictl is not installed, it will fall back to docker CLI." Do you mean change cluster e2e test to use crictl when health status checking?

**Comment by Random-Liu** [comment]
@mikebrow Some e2e test use those util functions, such as [here](https://github.com/kubernetes/kubernetes/blob/dca85f046fad9282637ba211bd60d8b04ad9f72c/test/e2e/storage/persistent_volumes-disruptive.go#L190).

**Comment by Random-Liu** [comment]
Discussed with @yujuhong offline. She brought up a very good point.
Today CRI is not versioned yet. Using `crictl` in our test environment means that once we make a breaking change to CRI, the test using `crictl` will fail.

We should avoid that.

Actually for this particular change, we can use `ps` and `kill` directly which is general for all container runtime.

I'll make the change.

**Comment by Random-Liu** [comment]
Remove LGTM for now in favor of https://github.com/kubernetes/kubernetes/pull/58024#issuecomment-356791939. @feiskyer 

Addressed comments. @yujuhong

**Comment by feiskyer** [comment]
> Today CRI is not versioned yet. Using crictl in our test environment means that once we make a breaking change to CRI, the test using crictl will fail.

Agreed. LGTM

File: test/e2e/framework/util.go [hunk 1]
```diff
@@ -3900,9 +3900,7 @@ func sshRestartMaster() error {
 	}
 	var command string
 	if ProviderIs("gce") {
-		// `kube-apiserver_kube-apiserver` matches the name of the apiserver
-		// container.
-		command = "sudo docker ps | grep kube-apiserver_kube-apiserver | cut -d ' ' -f 1 | xargs sudo docker kill"
+		command = "sudo ps -ef | grep \"[k]ube-apiserver\" | awk '{print $2}' | xargs sudo kill"
```
Comment (yujuhong) [inline test/e2e/framework/util.go]: Why not just use `pidof`?

Comment (yujuhong) [reply test/e2e/framework/util.go]: Also, it might be nice to factor this out as a function. Not required by this PR though.

Comment (Random-Liu) [reply test/e2e/framework/util.go]: Will do.

**Pull request merged**